chemprop/train/loss_functions.py

```diff
@@ -0,0 +1,202 @@
+from typing import Callable
+
+import torch
+import torch.nn as nn
+
+from chemprop.args import TrainArgs
+
+
+def get_loss_func(args: TrainArgs) -> Callable:
+    """
+    Gets the loss function corresponding to a given dataset type.
+
+    :param args: Arguments containing the dataset type ("classification", "regression", or "multiclass").
+    :return: A PyTorch loss function.
+    """
+
+    # Nested dictionary of the form {dataset_type: {loss_function: loss_function callable}}
+    # Note f1 loss function has been disabled because mcc is similar but less sensitive to unbalanced datasets
```

Comment:
If choosing to disable f1 here, should the functions below also be removed, and should it be removed from other files (e.g. args.py)? Or would it be better to allow it to be used with a warning/recommendation to use mcc instead?

Reply:
It is still valid as a metric so I'm leaving it as a metric in args. I don't believe it's available in args as a loss function though.

I'm on the fence about removing it altogether from the loss function list. It was not insubstantial to figure out how to do it in the first place, before I had second thoughts about including it as a loss function. My thinking was that if anyone wanted to implement it later, this would allow them to easily. Now thinking about that weighted against having isolated code that wouldn't get updated with other functions, I think you're right. I'll go back and remove it.

Reply:
Change made